### PR TITLE
feat: desyn adapter added

### PR DESCRIPTION
**Pull request by santinoYoung**
**NOTE**

### Please enable "Allow edits by maintainers" while putting up the PR.
____

## Type of change
- [x] New adapters
- [ ] Adapter-related bug fix

Description:

___

### Description

Please provide the following information based on the type of points calculation:

#### For TVL Points Calculation:

1. Dapp Description:
   
   DeSyn Protocol is an innovative decentralized asset management infrastructure on Web 3, empowering users to securely and transparently create and manage customized pool-based portfolios with various on-chain assets (tokens, NFTs, derivatives, etc.) via smart contract. Its current main product is a new type of farming launchpad for every project, featuring various structured products.

2. Criteria for Calculating User Points:
   

Users earn 1 desyn point per hour for depositing tokens equivalent to 1 ETH and 20 desyn points per hour for depositing tokens equivalent to 1 BTC.

3. Collateral Asset Locking Contract:
   
   > **Note**: If there are multiple `tokenAddress`, please specify which contracts the different `tokenAddress` are locked in.

 Collateral Asset Locking Contract：code Item.PoolAddress
5. LP Contract:
   
   > **Note**: User balance is calculated as LP percentage multiplied by total collateral.

LP Contract : code Item.LpAddress

### CODE REVIEW

**Comment by xsteadybcgo** [comment]
@santinoYoung I have updated the form at the top. Could you please add the relevant information about Desyn? This will help us in reviewing the PR.

**Comment by xsteadybcgo** [comment]
https://api.desyn.io/zklink/points_gateway/points/get_zklink_balance_list
@santinoYoung 404 error when fetch this endpoint

File: src/adapters/aqua/execution/package-lock.json
Comment (xsteadybcgo) [inline src/adapters/aqua/execution/package-lock.json:1]: do not delete this file

File: src/adapters/desyn/execution/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+import { UserTVLData } from './sdk/types';
+import { getUserTVLData } from './sdk/lib';
+
+
+export const getUserTransactionData = async (blockNumber: number): Promise<UserTVLData[]> => {
+  return await getUserTVLData(blockNumber);
+};
+
```
Comment (xsteadybcgo) [inline src/adapters/desyn/execution/src/index.ts:8]: it should export a function named `getUserTVLData`

File: src/adapters/desyn/execution/src/sdk/lib.ts [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+export const getTimestampAtBlock = async (blockNumber: number) => {
+  const provider = new JsonRpcProvider("https://rpc.zklink.io");
+  const block = await provider.getBlock(blockNumber);
+  return Number(block?.timestamp);
+};
+  
+export async function getUserTVLData(blockNumber: number):Promise<UserTVLData[]>  {
+
+  const response = (await axios.get("https://api.desyn.io/zklink/points_gateway/points/get_zklink_balance_list?blockNumber=" + blockNumber));
+  const data = response.data.list;
```
Comment (xsteadybcgo) [inline src/adapters/desyn/execution/src/sdk/lib.ts:14]: Unable to request successfully, 404 error

Comment (santinoYoung) [reply src/adapters/desyn/execution/src/sdk/lib.ts:14]: The api is reconditioned as GET

File: src/adapters/desyn/execution/src/index.ts (hunk 1 above)
Comment (santinoYoung) [reply src/adapters/desyn/execution/src/index.ts:8]: I resubmitted it.  export  function getUserTVLData

File: src/adapters/desyn/execution/src/sdk/lib.ts (hunk 2 above)
Comment (xsteadybcgo) [reply src/adapters/desyn/execution/src/sdk/lib.ts:14]: I encountered a 500 error. Could you ensure the stability of the service? It's crucial for retrieving points accurately.
do you have the resources to develop a subgraph as a data source?

Comment (santinoYoung) [reply src/adapters/desyn/execution/src/sdk/lib.ts:14]: I will keep the service stable, for the time being, we can only provide data through api form.

File: src/adapters/aqua/execution/package-lock.json
Comment (xsteadybcgo) [inline src/adapters/aqua/execution/package-lock.json:1]: Please do not delete this file. @santinoYoung

File: src/adapters/desyn/execution/src/sdk/lib.ts [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+
+  const userTxData: UserTVLData[] = data.map((item: any) => {
+    return {
+      blockNumber: blockNumber,
+      userAddress: item.userAddress,
+      poolAddress: item.poolAddress,
+      lpAddress: item.lpAddress,
+      tokenAddress: item.tokenAddress,
+      balance: item.balance,
```
Comment (xsteadybcgo) [inline src/adapters/desyn/execution/src/sdk/lib.ts:23]: The `balance` needs to return the raw balance. Please ensure that the `decimals` are preserved in the returned data

**Review by xsteadybcgo — COMMENTED** [review]
LGTM

File: src/adapters/aqua/execution/package-lock.json
Comment (santinoYoung) [reply src/adapters/aqua/execution/package-lock.json:1]: ok, I'll reload it based on the main branch.

File: src/adapters/desyn/execution/src/sdk/lib.ts (hunk 3 above)
Comment (xsteadybcgo) [reply src/adapters/desyn/execution/src/sdk/lib.ts:23]: @santinoYoung Sorry for the misunderstanding. What I meant was that the `balance` needs to be returned as an integer.

Comment (santinoYoung) [reply src/adapters/desyn/execution/src/sdk/lib.ts:23]: > @santinoYoung Sorry for the misunderstanding. What I meant was that the `balance` needs to be returned as an integer.

Sorry, I don't quite understand, now all that is returned is the integer and decimals precision

Comment (santinoYoung) [reply src/adapters/desyn/execution/src/sdk/lib.ts:23]: For example return data 
`{
        "userAddress": "0x985925d21B12aB4a148bd794b9Af46faFD7CeE0E",
        "tokenAddress": "0xfa70fD01EbD5aa64f31E5D3575C444919C79275E",
        "lpAddress": "0xe116d8dCcA2aA97150Aa24fa4957859760AEC6fB",
        "poolAddress": "0x1ccc2353d0cC1C0B201Cf4566141188cD6F335d6",
        "balance": "1000000000000000",
        "timestamp": 1723716235,
        "decimals": 18
      }`

Comment (xsteadybcgo) [reply src/adapters/desyn/execution/src/sdk/lib.ts:23]: @santinoYoung  excute the command `npm run adapter:tvl -- desyn tvl 5228742`
image

Comment (xsteadybcgo) [reply src/adapters/desyn/execution/src/sdk/lib.ts:23]: @santinoYoung run the command `npm run adapter:tvl -- desyn tvl 5260011`
image

Comment (santinoYoung) [reply src/adapters/desyn/execution/src/sdk/lib.ts:23]: > npm run adapter:tvl -- desyn tvl 5260011

Oh. I got it, thanks. I'll take care of it.

Comment (santinoYoung) [reply src/adapters/desyn/execution/src/sdk/lib.ts:23]: It‘s fixed it～

**Review by xsteadybcgo — APPROVED** [review]

**Pull request merged**